### PR TITLE
ARROW-5466: [Java][CI] Dockerize Java CI, run all JDK builds in single Travis entry

### PR DESCRIPTION
**Pull request by wesm**
Since OpenJDK9 has been superseded by OpenJDK11, it is not available in package repositories, so I'm not sure it's worth maintaining a build for this. 

I have pushed a pre-built Docker image for this to

https://cloud.docker.com/u/ursalab/repository/docker/ursalab/arrow-ci-java-all-jdks

### CODE REVIEW

**Comment by wesm** [comment]
Well that was relatively painless

**Review by xhochy — APPROVED** [review]
+1, LGTM

File: docker-compose.yml [hunk 1]
```diff
@@ -292,6 +292,18 @@ services:
       - .:/arrow:ro  # ensures that docker won't contaminate the host directory
       - maven-cache:/root/.m2:delegated
 
+  java-all-jdks:
+    # Usage:
+    #   docker-compose build java-all-jdks
+    #   docker-compose run java-all-jdks
+    image: ursalab/arrow-ci-java-all-jdks:latest
```
Comment (kou) [inline docker-compose.yml]: Should we use `ursalabs` (the last `s` is missing) for organization name on Docker Hub?

File: java/Dockerfile.all-jdks [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# under the License.
+
+FROM ubuntu:18.04
+
+# install build essentials
+RUN export DEBIAN_FRONTEND=noninteractive && \
+    apt-get update -y -q && \
+    apt-get install -y -q --no-install-recommends \
+        wget software-properties-common gpg-agent && \
+    apt-add-repository -y ppa:webupd8team/java && \
```
Comment (kou) [inline java/Dockerfile.all-jdks]: We can install both `openjdk-8-jdk` and `openjdk-11-jdk` without this:

  * https://packages.ubuntu.com/bionic/openjdk-8-jdk
  * https://packages.ubuntu.com/bionic/openjdk-11-jdk

File: ci/docker_build_java.sh [hunk 3]
```diff
@@ -26,9 +26,14 @@ mkdir -p /build/java
 arrow_src=/build/java/arrow
 
 pushd /arrow
-  rsync -a header java format integration $arrow_src
+rsync -a header java format integration $arrow_src
```
Comment (kou) [inline ci/docker_build_java.sh]: How about adding `--delete` `rsync` option or removing `${arrow_src}` before `rsync`?
If there are some built files with OpenJDK 8, OpenJDK 11 build may be conflicted.

File: docker-compose.yml (hunk 1 above)
Comment (wesm) [reply docker-compose.yml]: It seems that this name is already taken on Docker Hub. I don't know who owns it

File: ci/docker_build_java.sh (hunk 3 above)
Comment (wesm) [reply ci/docker_build_java.sh]: OK, will do

File: java/Dockerfile.all-jdks (hunk 2 above)
Comment (wesm) [reply java/Dockerfile.all-jdks]: OK, thanks

File: docker-compose.yml (hunk 1 above)
Comment (kou) [reply docker-compose.yml]: Oh. I got it.

**Review by pravindra — APPROVED** [review]
lgtm

**Comment by wesm** [comment]
Will merge this once the build runs

File: .travis.yml [hunk 4]
```diff
@@ -199,32 +199,14 @@ matrix:
...
-    os: linux
-    jdk: openjdk9
-    before_script:
-    - if [ $ARROW_CI_JAVA_AFFECTED != "1" ]; then exit; fi
-    script:
-    - $TRAVIS_BUILD_DIR/ci/travis_script_java.sh
-  - name: "Java w/ OpenJDK 11"
-    language: java
+  - name: "Java OpenJDK8 and OpenJDK11"
+    language: cpp
```
Comment (BryanCutler) [inline .travis.yml]: Should language be Java?

File: ci/docker_build_java.sh [hunk 5]
```diff
@@ -25,10 +25,18 @@ mkdir -p /build/java
...
+rm -rf $arrow_src
+
 pushd /arrow
-  rsync -a header java format integration $arrow_src
+rsync -a header java format integration $arrow_src
 popd
 
 pushd $arrow_src/java
-  mvn -B -DskipTests -Drat.skip=true install
+mvn -B -DskipTests -Drat.skip=true install
```
Comment (BryanCutler) [inline ci/docker_build_java.sh]: Just to confirm, this is build only and you don't want tests run?

**Review by BryanCutler — APPROVED** [review]
Looks good, I just had a couple minor questions

File: .travis.yml (hunk 4 above)
Comment (wesm) [reply .travis.yml]: This one doesn't matter b/c it's Java

File: ci/docker_build_java.sh (hunk 5 above)
Comment (wesm) [reply ci/docker_build_java.sh]: Ah, thanks for catching. Yeah we need to run the tests. I'll fix

**Pull request closed without merging**